src/subtype.c

```diff
@@ -469,9 +469,15 @@ static int obviously_disjoint(jl_value_t *a, jl_value_t *b, int specificity)
 // compute a least upper bound of `a` and `b`
 static jl_value_t *simple_join(jl_value_t *a, jl_value_t *b)
 {
-    if (a == jl_bottom_type || b == (jl_value_t*)jl_any_type || obviously_egal(a,b))
+    if (a == (jl_value_t *)jl_any_type ||
+        b == (jl_value_t *)jl_any_type ||
+        //TODO: Add a fastpath for this check.
+        jl_subtype((jl_value_t *)jl_any_type, a) ||
```

Comment:
`jl_subtype` might be rather glitchy here, since `a` or `b` might have free typevars. Where are the weird Any's getting introduced into the code wrapped?